specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,963 @@
+on fixed-point numbers via intrinsics. Front-ends can use any integer or floating
+point width when generating code, and it is the responsibility of a backend to
+provide a compile-time error message if the frontend uses an integer width greater
+than that which is supported.
+
+When an adaptive profile program indicates that it is using a particular data
+type, then instruction set (`qis`) functions or classical (`rt`) functions that
+a backend may support can now be called. For example, consider that if a backend
+supports integer computations and random number generation, then an adaptive profile
+qir program may have code like the following to do randomized benchmarking:
+
+```llvm
+%0 = call i64 @__quantum__rt__rand_range(i64 0, i64 2)
+%1 = icmp eq i64 %0, 0
+br i1 %1, label %zero_rand_sequence, label %one_rand_sequence
+```
+
+By combining mid-circuit measurements with instructions on classical
+data types, you can conditionally apply gates based on logic using multiple
+mid-circuit measurements and boolean computations:
+
+```llvm
+  tail call void @__quantum__qis__h__body(%Qubit* null)
+  tail call void @__quantum__qis__mz__body(%Qubit* null, %Result* null)
+  tail call void @__quantum__qis__reset__body(%Qubit* null)
+  %0 = tail call i1 @__quantum__rt__read_result__body(%Result* null)
+  tail call void @__quantum__qis__h__body(%Qubit* nonnull inttoptr (i64 1 to %Qubit*))
+  tail call void @__quantum__qis__mz__body(%Qubit* nonnull inttoptr (i64 1 to %Qubit*), %Result* nonnull inttoptr (i64 1 to %Result*))
+  tail call void @__quantum__qis__reset__body(%Qubit* nonnull inttoptr (i64 1 to %Qubit*))
+  %1 = tail call i1 @__quantum__rt__read_result__body(%Result* nonnull inttoptr (i64 1 to %Result*))
+  %2 = and i1 %0, %1
+  br i1 %2, label %then, label %continue
+
+then:                                   ; preds = %entry
+  tail call void @__quantum__qis__x__body(%Qubit* nonnull inttoptr (i64 2 to %Qubit*))
+  br label %continue__1.i.i.i
+
+continue:
+...
+```
+
+An adaptive profile program must indicate which classical data types must be supported to execute it. This is done by
+setting the following module flags to indicate the widths of various classical types as values for the rightmost
+section of the module flag metadata, like in the table below. Any program like the
+previous one that uses instructions on classical data types must have the
+appropriate module flag set, or it is not a legal adaptive profile program. A module flag can indicate that the
+program uses multiple classical types by adding multiple "end" fields in the module flags, each an i32 representing
+a type's width. For example, `!{i32 1, !"classical_ints", i32 16, i32 32}` indicates that both 16-bit integers
+and 32-bit integers are used in the adaptive profile program. On the other hand, `!{i32 1, !"classical_ints", i32 0}` indicates
+that the program does not use classical integer computations.
```

Comment:
If an extension to the core spec isn't supported, I think omitting the flag makes more sense than having to interpret the value as a special case.